File: src/main/java/com/pidev/welend/services/InsuranceTransactionService.java
package com.pidev.welend.services;

import com.pidev.welend.entities.insurance;
import com.pidev.welend.entities.insuranceTransaction;

import java.util.HashMap;
import java.util.List;

public interface InsuranceTransactionService {
    public insuranceTransaction addInsuranceTransaction(insuranceTransaction i);
    public insuranceTransaction updateInsuranceTransaction(insuranceTransaction i);
    public List<insuranceTransaction> getAllInsuranceTransaction();
    public insuranceTransaction getInsuranceTransactionById(Integer insuranceTransactionID);
    public void deleteInsuranceTransaction(Integer insuranceTransactionId);
    public List<insuranceTransaction> getInsuranceTransactionByInsuranceID(Integer insuranceID);
    public HashMap<insuranceTransaction, String> checkAllUnpaidInsuranceTransactionByYear(Integer insuranceID,Integer year);
    List<insuranceTransaction> getAllInsuranceTransactionPendingByInsurance(Integer insuranceID);
    List<insuranceTransaction> getAllInsuranceTransactionNotfullysetteledByInsurance(Integer insuranceID);
    List<insuranceTransaction> getAllInsuranceTransactionSettledByInsurance(Integer insuranceID);
    List<insuranceTransaction> getAllInsuranceTransactionByAcountID(Integer accountID);
}

File: WeLend Front/src/app/services/insuranceTransaction/insurance-transaction.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Insurance} from "../../entities/insurance";
import {InsuranceTransaction} from "../../entities/insuranceTransaction";
import {environment} from "../../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class InsuranceTransactionService {

  private apiUrl =environment.apiBaseUrl;

  constructor(private http: HttpClient) {}
  public addInsuranceTransaction(insuranceTransaction:InsuranceTransaction): Observable<InsuranceTransaction>{
    return this.http.post<InsuranceTransaction>(`${this.apiUrl}/insuranceTransaction/add`,insuranceTransaction);
  }
  public updateInsuranceTransaction(insuranceTransaction:InsuranceTransaction): Observable<InsuranceTransaction>{
    return this.http.put<InsuranceTransaction>(`${this.apiUrl}/insuranceTransaction/update`,insuranceTransaction);
  }
  public deleteInsuranceTransaction(insuranceTransactionID:number): Observable<void>{
    return this.http.delete<void>(`${this.apiUrl}/insurance/delete/`+insuranceTransactionID);
  }
  public getAllInsuranceTransaction(): Observable<InsuranceTransaction[]>{
    return this.http.get<InsuranceTransaction[]>(`${this.apiUrl}/insuranceTransaction/getAll`);
  }
  public getInsuranceTransactionById(insuranceTransactionID:number): Observable<InsuranceTransaction>{
    return this.http.get<InsuranceTransaction>(`${this.apiUrl}/insuranceTransaction/getByID/`+insuranceTransactionID);
  }
  public getInsuranceTransactionByInsuranceId(insuranceID:number): Observable<InsuranceTransaction[]>{
    return this.http.get<InsuranceTransaction[]>(`${this.apiUrl}/insuranceTransaction/getByInsuranceID/`+insuranceID);
  }
  public checkAllUnpaidInsuranceTransactionByYear(insuranceID:number,year:number): Observable<InsuranceTransaction[]>{
    return this.http.get<InsuranceTransaction[]>(`${this.apiUrl}/insuranceTransaction/checkAllUnpaidInsuranceTransactionByYear/`+insuranceID+'/'+year);
  }
  public getAllInsuranceTransactionPendingByInsurance(insuranceID:number): Observable<InsuranceTransaction[]>{
    return this.http.get<InsuranceTransaction[]>(`${this.apiUrl}/insuranceTransaction/getAllInsuranceTransactionPendingByInsurance/`+insuranceID);
  }
  public getAllInsuranceTransactionNotfullySetteledByInsurance(insuranceID:number): Observable<InsuranceTransaction[]>{
    return this.http.get<InsuranceTransaction[]>(`${this.apiUrl}/insuranceTransaction/getAllInsuranceTransactionNotfullySetteledByInsurance/`+insuranceID);
  }
  public getAllInsuranceTransactionSettledByInsurance(insuranceID:number): Observable<InsuranceTransaction[]>{
    return this.http.get<InsuranceTransaction[]>(`${this.apiUrl}/insuranceTransaction/getAllInsuranceTransactionSettledByInsurance/`+insuranceID);
  }
  public getAllInsuranceTransactionByAccountID(accountID:number): Observable<InsuranceTransaction[]>{
    return this.http.get<InsuranceTransaction[]>(`${this.apiUrl}/insuranceTransaction/getAllInsuranceTransactionByAccountID/`+accountID);
  }

}
